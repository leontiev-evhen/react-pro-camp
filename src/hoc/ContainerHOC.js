import React from 'react';
import PropTypes from 'prop-types';
import { Preloader } from '../components';

function ContainerHOC(Component) {
	function containerHOC({ isFetching, error, ...props }) {
		if (isFetching) return <Preloader />;
		if (error) return <div className="error-message">{error}</div>;
		return <Component {...props} />;
	}
	containerHOC.propTypes = {
		isFetching: PropTypes.bool.isRequired,
		error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]).isRequired,
	};
	return containerHOC;
}

export default ContainerHOC;
