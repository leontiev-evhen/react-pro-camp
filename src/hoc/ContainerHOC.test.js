import React from 'react';
import { shallow } from 'enzyme';
import ContainerHOC from './ContainerHOC';

describe('ContainerHOC', () => {
	const SomeComponent = () => <h1>Some Component</h1>;
	const DivContainerHOC = ContainerHOC(SomeComponent);
	describe('ContainerHOC Preloader', () => {
		const props = {
			isFetching: true,
			error: '',
		};
		const Component = shallow(<DivContainerHOC {...props} />);
		it('Render Preloader', () => {
			expect(Component.find('Preloader')).toHaveLength(1);
		});
		it('Render correctly Preloader component', () => {
			expect(Component).toMatchSnapshot();
		});
	});
	describe('ContainerHOC error', () => {
		const props = {
			isFetching: false,
			error: 'Unknown error',
		};
		const Component = shallow(<DivContainerHOC {...props} />);
		it('Render error', () => {
			expect(Component.find('.error-message').text()).toEqual(props.error);
		});
		it('Render correctly error', () => {
			expect(Component).toMatchSnapshot();
		});
	});
	describe('ContainerHOC Component', () => {
		const props = {
			isFetching: false,
			error: '',
		};
		const Component = shallow(<DivContainerHOC {...props} />);
		it('Render Component', () => {
			expect(Component.find(SomeComponent)).toBeTruthy();
		});
		it('Render correctly Component', () => {
			expect(Component).toMatchSnapshot();
		});
	});
});
