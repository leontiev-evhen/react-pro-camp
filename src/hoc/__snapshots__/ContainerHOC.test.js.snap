// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerHOC ContainerHOC Component Render correctly Component 1`] = `<SomeComponent />`;

exports[`ContainerHOC ContainerHOC Preloader Render correctly Preloader component 1`] = `<Preloader />`;

exports[`ContainerHOC ContainerHOC error Render correctly error 1`] = `
<div
  className="error-message"
>
  Unknown error
</div>
`;
